cql3/statements/select_statement.cc

```diff
@@ -1442,6 +1442,50 @@ indexed_table_select_statement::find_index_partition_ranges(query_processor& qp,
     }));
 }
 
+indexed_table_select_statement::partition_ranges_generator
+indexed_table_select_statement::make_index_partition_ranges(query_processor& qp, service::query_state& state, const query_options& options) const
+{
+    std::unique_ptr<cql3::query_options> internal_options = std::make_unique<cql3::query_options>(cql3::query_options(options));
+    auto paging_state = options.get_paging_state();
+    int user_page_size = options.get_page_size();
+    // max internal page size
+    // Hardcoded to 2000 so that contiguous allocations from std::vector<partition_range> don't exceed 256 KiB (sizeof(dht::partition_range) == 120).
+    // Such vectors are returned by `find_index_partition_ranges()`.
+    int max_page_size = internal_paging_size / 5;
+    bool is_unpaged = user_page_size < 0;
```

Comment:
I'd prefer reversing the logic and use `is_paged` rather than `is_unpaged`.
It should be less confusing this way.